typings.d.ts

```diff
@@ -1,627 +1,25 @@
-      }
-    }
-    scheduledFor: {
-      datetime: Date
-    }
-    action: ScheduledActionActionType
-  }
-
-  /* Dialogs API */
-
-  interface OpenAlertOptions {
-    title: string
-    message: string
-    confirmLabel?: string
-    shouldCloseOnEscapePress?: boolean
-    shouldCloseOnOverlayClick?: boolean
-  }
-
-  type OpenConfirmOptions = OpenAlertOptions & {
-    cancelLabel?: string
-    intent?: 'primary' | 'positive' | 'negative'
-  }
-
-  interface OpenCustomWidgetOptions {
-    id?: string
-    width?: number | 'small' | 'medium' | 'large' | 'fullWidth'
-    minHeight?: number | string
-    allowHeightOverflow?: boolean
-    position?: 'center' | 'top'
-    title?: string
-    shouldCloseOnOverlayClick?: boolean
-    shouldCloseOnEscapePress?: boolean
-    parameters?: Object
-  }
-
-  interface DialogsAPI {
-    /** Opens a simple alert window (which can only be closed). */
-    openAlert: (options: OpenAlertOptions) => Promise<boolean>
-    /** Opens a confirmation window. A user can either confirm or cancel the dialog. */
-    openConfirm: (options: OpenConfirmOptions) => Promise<boolean>
-    /** Opens a prompt window. A user can either provide a string input or cancel the dialog. */
-    openPrompt: (
-      options: OpenConfirmOptions & {
-        defaultValue?: string
-      }
-    ) => Promise<string | boolean>
-    /** Opens an extension in a dialog. */
-    openExtension: (options: OpenCustomWidgetOptions) => Promise<any>
-    /** Opens the current app in a dialog */
-    openCurrentApp: (options?: Omit<OpenCustomWidgetOptions, 'id'>) => Promise<any>
-    /** Opens a dialog for selecting a single entry. */
-    selectSingleEntry: <T = Object>(options?: {
-      locale?: string
-      contentTypes?: string[]
-    }) => Promise<T | null>
-    /** Opens a dialog for selecting multiple entries. */
-    selectMultipleEntries: <T = Object>(options?: {
-      locale?: string
-      contentTypes?: string[]
-      min?: number
-      max?: number
-    }) => Promise<T[] | null>
-    /** Opens a dialog for selecting a single asset. */
-    selectSingleAsset: <T = Object>(options?: {
-      locale?: string
-      mimetypeGroups?: string[]
-    }) => Promise<T | null>
-    /** Opens a dialog for selecting multiple assets. */
-    selectMultipleAssets: <T = Object>(options?: {
-      locale?: string
-      min?: number
-      max?: number
-      mimetypeGroups?: string[]
-    }) => Promise<T[] | null>
-  }
-
-  /* Navigator API */
-
-  interface NavigatorAPIOptions {
-    /** use `waitForClose` if you want promise to be resolved only after slide in editor is closed */
-    slideIn?: boolean | { waitForClose: boolean }
-  }
-
-  interface PageExtensionOptions {
-    /** If included, you can navigate to a different page extension. If omitted, you will navigate within the current extension. */
-    id?: string
-    /** Navigate to a path withing your page extension. */
-    path?: string
-  }
-
-  interface NavigatorSlideInfo {
```

Comment:
I think we used some of these types in `field-editors` even if was not exposed. How we make sure that it properly and doesn't introduce a breaking change?

Reply:
It's still defined in `lib/types.ts` so internal files can import it from there.

Reply:
let's test it with field-editors before merging and releasing. ok?

Reply:
It turns out that typescript was exposing all of these types even without `export` since they're inside the module declaration so I'll expose them again :+1:

Reply:
With `lib/types.ts` being included in the package and referenced from `typings.d.ts`, typescript would suggest imports from `contentful-ui-extensions-sdk/lib/types`. To avoid this, I've added a `typings` npm script to generate a standalone `typings.d.ts` file in https://github.com/contentful/ui-extensions-sdk/pull/291/commits/cf21556ea7b68e7b0dd118f233128153e2be8242